prober/utils.go

```diff
@@ -26,6 +27,28 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+var protocolToGauge = map[string]float64{
+	"ip4": 4,
+	"ip6": 6,
+}
+
+type resolver struct {
```

Comment:
This seems unnecessary, we only call one method of it.

Reply:
Yes but the logic in `resolve` is used twice. I can have a function that takes ctx, resolver, target, protocol and eliminate this wrapper, but it seems like this is too much parameters.

Please let me know if keeping this wrapper is acceptable.

Reply:
This type appears unneeded. Also I don't see an issue with duplicating 4 lines of code.